src/utils/RandomMatching.ts

```diff
@@ -0,0 +1,182 @@
+import { io } from 'socket.io-client';
+import type { Socket } from 'socket.io-client';
+import type { MatchingInformation } from 'types/matching';
+import { MATCHING_SOCKET_EVENT, EXCEPTION_MESSAGE } from 'constants/matching';
+
+export class RandomMatching {
+  public socket: Socket | null = null;
+
+  public peer: RTCPeerConnection | null = null;
+
+  private audioStream: MediaStream | null = null;
+
+  private async canUseAudio() {
+    try {
+      this.audioStream = await navigator.mediaDevices.getUserMedia({
+        audio: true,
+      });
+
+      const audioPermissionStatus = await navigator.permissions.query({
+        name: 'microphone' as PermissionName,
+      });
+
+      const canUse = audioPermissionStatus.state !== 'denied';
+
+      return canUse;
+    } catch (error) {
+      return false;
+    }
+  }
+
+  public async startMatching({
+    userInformation,
+    onSuccess,
+    onError,
+  }: {
+    userInformation: { id: string; username: string };
```

Comment:
p3;
대체로 user라고 작성해도 사용자의 정보라는 것을 알 수 있다고 생각해서, information 제거해도 좋을 것 같아요!

```suggestion
    user: Pick<User, 'id' | 'username'>;
```

Reply:
수정하였습니다!

작업 커밋 : 1236ddf